kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ElementAtEvaluator.java

```diff
@@ -101,20 +102,31 @@ public boolean isNullAt(int rowId) {
             @Override
             public String getString(int rowId) {
                 lookupValue(rowId);
-                return lastLookupValue == null ? null : (String) lastLookupValue;
+                return lastLookupValue == null ? null : lastLookupValue;
             }
 
             private Object lookupValue(int rowId) {
                 if (rowId == lastLookupRowId) {
                     return lastLookupValue;
                 }
-                // TODO: this needs to be updated after the new way of accessing the complex
-                //  types is merged.
                 lastLookupRowId = rowId;
                 String keyValue = lookupKey.getString(rowId);
-                lastLookupValue = map.getMap(rowId).get(keyValue);
+                lastLookupValue = findValueForKey(map.getMap(rowId), keyValue);
                 return lastLookupValue;
             }
+
+            private String findValueForKey(MapValue mapValue, String key) {
+                ColumnVector keyVector = mapValue.getKeys();
+                for (int i = 0; i < mapValue.getSize(); i++) {
+                    if ((keyVector.isNullAt(i) && key == null) ||
+                            (!keyVector.isNullAt(i) && keyVector.getString(i).equals(key))) {
+                        return mapValue.getValues().isNullAt(i) ? null :
+                                mapValue.getValues().getString(i);
+                    }
+                }
+                // TODO what is the behavior for element_at if not in the map?
```

Comment:
@vkorukanti I can update the docs for the function as well

Reply:
We shouldn't return null. We should return an `Optional`.

What if the value for the key is `null`? Isn't that perfectly valid?

Reply:
Thank you! Please add the docs. I didn't mention the `null` behavior. The behavior should be it should return `null`. We have tests that confirm this.